### PR TITLE
Add information about build folder for beginners

### PR DESCRIPTION
**Pull request by pxalcantara**
Some suggestions for more detailed information for beginner users about the build folder and how to run just a single test. I also had problems with these points in my first experience.

### CODE REVIEW

**Comment by pxalcantara** [comment]
I get your point, why don't do exactly like you said? 
Using `colcon` this is the path ...
Using `CMake` this is the path ...
It would be too confusing?

**Comment by chapulina** [comment]
I think that's one way, but I'm not sure if we should also restructure some of our other documentation to also be split into colcon / vanilla cmake :thinking: It could be a bit confusing if this is the only place where we mention options. I think it's not clear to a new user that they have options.

**Comment by pxalcantara** [comment]
OK, what is best for the community, just let me know :wink:

**Comment by maryaB-osr** [comment]
@chapulina Hmm... If I'm not mistaken, we tell users to use colcon in our installation instructions, correct? In that case, I think it's ok to assume that they will have their workspace layout in the typical colcon path. Especially if we are concerned with beginner users; they will most likely have just followed the installation instructions verbatim. 

I think the bigger question here is: should we point out that using plain CMake is an option in the installation instructions? I feel like anyone who is experienced enough to use plain CMake will be able to identify that it's an option without being explicitly told so, and will also know that their package path will be different than the one setup by colcon. If that's a fair assumption, I think it's fair to **not** mention the CMake path to avoid confusing people who aren't familiar with it. Instead we can say something like:

> If you followed our source installation instructions/build instructions, the path to ign-math would look like this...

Or 

> Assuming you followed our installation instructions using colcon, the path to ign-math...

File: contributing.md [hunk 1]
```diff
@@ -204,7 +204,7 @@ get aquainted with this development process.
 
         sudo apt-get install cppcheck
 
-    To check your code, run the following script from your `build` folder :
+    To check your code, run the following script from the `build` folder of the project that you're working. If you're working in the *ignition-math*, for instance, the path for the folder should be something similar to this `~/citacel_ws/build/ignition-math6`, than, run the script inside this folder:
```
Comment (maryaB-osr) [inline contributing.md:207]: Just some corrections before we decide exactly what to say:
```suggestion
    To check your code, run the following script from the `build` folder of the project that you're working on. 
    If you're working on *ignition-math*, for instance, the path for the folder should be something similar to `~/citadel_ws/build/ignition-math6`.
    Then, run the script inside this folder:
```

**Comment by pxalcantara** [comment]
>  we tell users to use colcon in our installation instructions, correct? In that case, I think it's ok to assume that they will have their workspace layout in the typical colcon path. Especially if we are concerned with beginner users; they will most likely have just followed the installation instructions verbatim.

It was what I did. As I'm a beginner and don't know too much about CMake and compilation tools, I've always used the tool that is recommended in the tutorials, like `colcon` and `catkin`, for instance.

>  If you followed our source installation instructions/build instructions, the path to ign-math would look like this...
> 
> Or
> 
> Assuming you followed our installation instructions using colcon, the path to ign-math...

I really like this proposal, make clear that this is the condition that you have if you followed the instructions that are in the installation tutorial and, also, that there are other options.

**Comment by chapulina** [comment]
> we tell users to use colcon in our installation instructions, correct? 

Each library's README has instructions on how to compile it using plain CMake (as opposed to colcon). See [this example](https://github.com/ignitionrobotics/ign-rendering/#building-from-source). These READMEs will also be rendered on the main site, like [here](https://ignitionrobotics.org/libs/rendering).

> should we point out that using plain CMake is an option in the installation instructions?

Based on the answer above, we're already doing this. So the question becomes about whether we should change that.

> anyone who is experienced enough to use plain CMake will be able to identify that it's an option

I often agree with this sentiment, but I wouldn't assume this is true for this specific case. I can think of 2 groups of users who may not be well versed in CMake and could benefit from specific CMake instructions:

* People wanting to embed Ignition into other CMake projects
* Users only interested in building a single Ignition library from source

---

Having all this said, I'd propose:

* In this documentation, always assume the use of colcon, but always be explicit about this assumption. So I agree with your proposal of "_Assuming you followed our installation instructions using colcon..._"
* On all the READMEs, explain that cmake is good for building a single library, and point users to the high-level colcon install if they want to install more libs, and therefore use colcon. This would address https://github.com/ignitionrobotics/ign-gazebo/issues/124.

**Comment by pxalcantara** [comment]
@chapulina I really liked your proposal

File: contributing.md [hunk 2]
```diff
@@ -204,7 +204,9 @@ get aquainted with this development process.
 
         sudo apt-get install cppcheck
 
-    To check your code, run the following script from your `build` folder :
+    To check your code, run the following script from the `build` folder of the project that you're working on. 
+    If you're working on *ignition-math*, for instance, the path for the folder should be something similar to `~/citadel_ws/build/ignition-math6`.
+    Then, run the script inside this folder:
```
Comment (maryaB-osr) [inline contributing.md:209]:
```suggestion
    To check your code, run the following script from the `build` folder of the project that you're working on.
    If you're working on *ignition-math*, for instance, the path for the folder should be something similar to `~/citadel_ws/build/ignition-math6`.
    The path example is assuming you followed [our installation instructions](/docs/citadel/install) using colcon.
    
    Then, run the script inside this folder:
```

**Comment by maryaB-osr** [comment]
@chapulina So it seems like mentioning CMake options everywhere colcon-related instructions are mentioned is actually valuable. In another PR I'll comb through and try to add those, but also maybe explicitly describe the difference between (or even just the option of) single library builds vs. the "full" installation somewhere in the non-library docs.

**Comment by pxalcantara** [comment]
@chapulina @maryaB-osr thanks for the comments. I'd like to know more about these differences between *"single library builds vs. the "full" installation"

**Comment by maryaB-osr** [comment]
@pxalcantara Sure! Ignition is a set of libraries, and each library can be used on its own to fulfill whatever function it serves. The libraries are listed [here](https://ignitionrobotics.org/libs). The readme for each library contains its own installation instructions for people that just want to use that library. That's what @chapulina is saying in the comment above about CMake vs. colcon. CMake is used in all the individual library installation instructions because it's good for building single libraries, whereas colcon is a tool that makes it easier to build many libraries together by sorting out their dependencies so you don't have to. This is why the "full" installation instructions (the ones you find on [ignitionrobotics.org/docs/citadel](ignitionrobotics.org/docs/citadel) for example) use colcon. Following the installation instructions from what we call the "main" docs or the "high-level" docs (everything under [ignitionrobotics.org/docs](ignitionrobotics.org/docs)) gets you all the ignition libraries. 

So in my last comment, I was saying that we should explicitly explain basically what I just explained to you somewhere in our "high-level" docs, so then we can make the distinction between colcon-related instructions and CMake-related instructions throughout the "high-level" docs without having to explain why we're including CMake every time, and without confusing people. Because someone who just installed one library from its readme may still end up on [ignitionrobotics.org/docs](ignitionrobotics.org/docs) to read the "general" tutorials, contributing guide, etc. And someone who did the full installation might see references to CMake and become worried that they're missing some part of the docs because they've never seen that before.

I hope that answers your question; it might be a little high-level so if you were looking for a more technical explanation we can wait for @chapulina to correct me! :sweat_smile:

**Comment by pxalcantara** [comment]
@maryaB-osr Thank you very much it's all clear now :star_struck:

**Pull request merged**